install.sh

```diff
@@ -77,26 +77,49 @@ process_arguments() {
 verify_configuration_file() {
   log "Verifying the config file."
 
+  HAS_VALIDATION_ERR=
   # Make sure the config values are defined
   set +e
   INVALID_CONTENT=$(grep -o '^[^#]*' "${CONFIG_ABS_PATH}" | grep '<\|>')
   set -e
   ENVIRONMENT_NAME=$(grep 'environment_name' "${CONFIG_ABS_PATH}" | sed -nE 's/^.*"(.*)".*$/\1/p')
 
+  # check license and admin password
+  export POPULATED_LICENSE=$(grep -o '^[^#]*' "${CONFIG_ABS_PATH}" | grep 'bamboo_license')
+  export POPULATED_ADMIN_PWD=$(grep -o '^[^#]*' "${CONFIG_ABS_PATH}" | grep 'bamboo_admin_password')
+
   if [ "${#ENVIRONMENT_NAME}" -gt 25 ]; then
     log "The environment name '${ENVIRONMENT_NAME}' is too long(${#ENVIRONMENT_NAME} characters)."
     log "Please make sure your environment name is less than 25 characters"
-    exit 1
+    HAS_VALIDATION_ERR=1
   fi
 
   if [ -n "${INVALID_CONTENT}" ]; then
-    log "Configuration file '${CONFIG_ABS_PATH}' is not valid."
+    log "Configuration file '${CONFIG_ABS_PATH##*/}' is not valid."
     log "Terraform uses this file to generate customised infrastructure for '${ENVIRONMENT_NAME}' on your AWS account."
-    log "Please modify '${CONFIG_ABS_PATH}' using a text editor and complete the configuration. "
+    log "Please modify '${CONFIG_ABS_PATH##*/}' using a text editor and complete the configuration. "
     log "Then re-run the install.sh to deploy the infrastructure."
     echo
     log "${INVALID_CONTENT}"
-    exit 0
+    HAS_VALIDATION_ERR=1
+  fi
+
+  if [ -n "${POPULATED_LICENSE}" ];  then
+    if [ -n "${TF_VAR_bamboo_license}" ]; then
+      echo "Please provide license in config file, or export it to the environment variable 'TF_VAR_bamboo_license'."
+      HAS_VALIDATION_ERR=1
+    fi
+  fi
+
+  if [ -n "${POPULATED_ADMIN_PWD}" ];  then
+    if [ -n "${TF_VAR_bamboo_admin_password}" ]; then
+      echo "Please provide admin password in config file, or export it to the environment variable 'TF_VAR_bamboo_admin_password'."
+      HAS_VALIDATION_ERR=1
+    fi
+  fi
+
+  if [ ! -n "${HAS_VALIDATION_ERR}" ]; then
+    exit 1
```

Comment:
I would also `log "There was a problem with the configuration file. Aborting execution" "ERROR"` line above.

Reply:
I don't think this message adds much as we get to this condition only if we already caught some invalid condition earlier, and for each invalid condition we already had logged the proper message before. Here the only action is aborting the process.

Reply:
Previously the whole execution printed:

```
2021-12-20 20:26:12 [INFO] Terraform will use '/Users/abrokes/code/data-center-terraform/abrokes-config.tfvars' to install the infrastructure.
2021-12-20 20:26:12 [INFO] Verifying the config file.
```
I know it was due to a bad condition but mistakes will happen in scripting and it is good to give a good guide where the execution was interrupted. And printing it out as `ERROR` is also useful.

Reply:
Actually in case of any condition we log the specific case before turn the error flag to `true`. 
such as:
```
2021-12-20 20:26:12 [INFO] Terraform will use '/Users/abrokes/code/data-center-terraform/abrokes-config.tfvars' to install the infrastructure.
2021-12-20 20:26:12 [ERROR] Configuration file 'abrokes-config.tfvars' is not valid.
2021-12-20 20:26:12 [INFO] Please make sure your environment name is less than 25 characters.
```

And if there are more than one issue, all will list before exit.

Again, wrapping up all error condition with a final error message is not bad idea, but if we don't have that message still we won't miss much.